internal/controller/etcdcluster_controller.go

```diff
@@ -139,10 +139,22 @@ func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	if memberListResp != nil {
 		memberCnt = len(memberListResp.Members)
 	}
+	targetReplica := *sts.Spec.Replicas // Start with the current size of the stateful set
+
+	// TODO: finish the logic later
+	if int(targetReplica) != memberCnt {
+		// TODO: finish the logic later
+		if int(targetReplica) < memberCnt {
+			// a new added learner hasn't started yet
+
+			// re-generate configuration for the new learner member;
+			// increase statefulsets's replica by 1
+		} else {
+			// an already removed member hasn't stopped yet.
 
-	if memberCnt != len(healthInfos) {
-		// Only proceed when all members are healthy
-		return ctrl.Result{}, fmt.Errorf("memberCnt (%d) isn't equal to healthy member count (%d)", memberCnt, len(healthInfos))
```

Comment:
Note: It has already been covered below

https://github.com/etcd-io/etcd-operator/blob/9766229d846c215d44b93e41d1483abe03726a6f/internal/controller/etcdcluster_controller.go#L191-L205